exportContributors script fails

**Describe the bug**
As described in a PR template, I tried to update documentation but `/scripts/exportContributors.gradle` failed at line 49.


**To Reproduce**
Steps to reproduce the behavior:
1. execute  `./gradlew exportContributors && ./gradlew && ./copyDocs.sh` like PR template recommends.
2. See error:

```
> Task :exportContributors FAILED

FAILURE: Build failed with an exception.

* Where:
Script '/home/ewoks/docs-as-code/docToolchain-Ewoks-Fork/scripts/exportContributors.gradle' line: 49

* What went wrong:
Execution failed for task ':exportContributors'.
> java.io.FileNotFoundException: /home/ewoks/docs-as-code/docToolchain-Ewoks-Fork/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/build/docs/contributors/src/docs/manual/06_Frequently_asked_Questions.adoc 
(File name too long)

```

**Expected behavior**
Export contributors script shouldn't failed because of multiple stacked path bug 

**Screenshots**
![Screenshot from 2020-08-24 19-59-02](https://user-images.githubusercontent.com/1666500/91079474-5ebf9580-e644-11ea-90d2-d6183b4a57fb.png)


**Configuration**
 - OS: **Linux**

--- a/.ci.sh
+++ b/.ci.sh
@@ -79,7 +79,40 @@ create_doc () {
   echo "#        Create documentation              #"
   echo "#                                          #"
   echo "############################################"
-  ./gradlew && ./copyDocs.sh
+  ./gradlew --stacktrace && ./copyDocs.sh
+}
+
+publish_doc () {
+  # Take from and modified http://sleepycoders.blogspot.de/2013/03/sharing-travis-ci-generated-files.html
+  # ensure publishing doesn't run on pull requests, only when token is available and only on JDK11 matrix build and on master or a travisci test branch
+  if [ "$TRAVIS_PULL_REQUEST" == "false" ] && [ -n "$GH_TOKEN" ] && [ "$TRAVIS_JDK_VERSION" == "openjdk11" ] && { [ "$TRAVIS_BRANCH" == "travisci" ] || [ "$TRAVIS_BRANCH" == "master" ]; } ; then
+    echo "############################################"
+    echo "#                                          #"
+    echo "#        Publish documentation             #"
+    echo "#                                          #"
+    echo "############################################"
+    echo -e "Starting to update gh-pages\n"
+
+    #go to home and setup git
+    cd "$HOME"
+    git config --global user.email "travis@travis-ci.org"
+    git config --global user.name "Travis"
+
+    #using token clone gh-pages branch
+    git clone --quiet --branch=gh-pages "https://${GH_TOKEN}@github.com/${TRAVIS_REPO_SLUG}.git" gh-pages > /dev/null
+
+    #go into directory and copy data we're interested in to that directory
+    cd gh-pages
+    rm -rf ./*
+    cp -Rf "$TRAVIS_BUILD_DIR"/docs/* .
+
+    #add, commit and push files
+    git add -f .
+    git commit -m "Travis build $TRAVIS_BUILD_NUMBER pushed to gh-pages"
+    git push -fq origin gh-pages > /dev/null
+
+    echo -e "Done publishing to gh-pages.\n"
+  fi
 }
 
 cleaning
@@ -88,3 +121,4 @@ unit_tests
 integration_tests
 check_for_clean_worktree
 create_doc
+publish_doc

--- a/.travis.yml
+++ b/.travis.yml
@@ -6,17 +6,13 @@ jdk:
   - openjdk12
   - openjdk13
 
+git:
+  depth: false
+
 addons:
   apt:
     packages:
       - shellcheck
-
-git:
-  depth: 3000
-
-addons:
-  apt:
-    packages:
       - graphviz
 
 before_cache:
@@ -27,6 +23,13 @@ cache:
     - $HOME/.gradle/caches/
     - $HOME/.gradle/wrapper/
 
+install: skip
+
 script:
   - ./.ci.sh
   - find . -type f -exec grep -l '^#!/bin/bash' {} + | xargs shellcheck
+
+env:
+  global:
+    # use `travis env set GH_TOKEN XXX` to set token to publish contents to GitHub pages
+    secure: XsGE9f9tLzIdqzKVCf6HyULznWPgw2k3Ghve+TAACsmSu1N1CpuNUdjfMS/9IdDncoP+AWV2aYZUAKImE/WyBvvXrgy8n3VpcCe13efU3fEXRkPQNBIZZOL1+/P7jeVTjX4cTe2ptemtJ2d4oFBcslJAX/c4D5HSu2/GMAfhe74=

--- a/README.adoc
+++ b/README.adoc
@@ -3,6 +3,10 @@
 
 == docToolchain
 
+:url-ci-travis: https://travis-ci.org/doctoolchain/doctoolchain
+
+image:https://api.travis-ci.org/doctoolchain/doctoolchain.svg?branch=master[link={url-ci-travis}]
+
 image::https://hacktoberfest.digitalocean.com/assets/HF-full-logo-b05d5eb32b3f3ecc9b2240526104cf4da3187b8b61963dd9042fdc2536e4a76c.svg[float=left,width=200]
 
 *This project joined https://hacktoberfest.digitalocean.com/[Hacktoberfest]!
@@ -86,7 +90,7 @@ image::https://opencollective.com/doctoolchain/sponsor/0/avatar.svg["Sponsor", l
 === History of this project
 
 I just found an older https://twitter.com/RalfDMueller/status/668540860649349120[tweet] which reminded me of the roots and beginning of this project.
-It all began with two important parts. 
+It all began with two important parts.
 A script called asciidoc2confluence and some scripts to export diagrams from enterprise architect.
 
 The first one is linked in the tweet above and the first commit dates back to mid of 2015.

--- a/build.gradle
+++ b/build.gradle
@@ -75,4 +75,4 @@ apply from: 'scripts/exportOpenApi.gradle'
 
 // let's set a defaultTask for convenience
 //defaultTasks 'exportChangeLog','exportJiraIssues','asciidoctor'
-defaultTasks 'exportChangeLog', 'exportContributors', 'exportMarkdown', 'generateHTML', 'htmlSanityCheck'
+defaultTasks 'exportMarkdown', 'exportChangeLog', 'exportContributors', 'generateHTML', 'htmlSanityCheck'

--- a/scripts/AsciiDocBasics.gradle
+++ b/scripts/AsciiDocBasics.gradle
@@ -32,23 +32,23 @@ logger.info("\nGradle project Properties [${props.size()}]:\n${props}\n=========
 
 if (project.hasProperty('jiraUser') && project.hasProperty('jiraPass')) {
     logger.info("Found passed Jira credentials")
-    config.jira.credentials = "${project.getProperty('jiraUser')}:${project.getProperty('jiraPass')}".bytes.encodeBase64().toString() 
+    config.jira.credentials = "${project.getProperty('jiraUser')}:${project.getProperty('jiraPass')}".bytes.encodeBase64().toString()
 }
 if (project.hasProperty('confluenceUser') && project.hasProperty('confluencePass')) {
     logger.info("Found passed Confluence credentials")
-    config.confluence.credentials = "${project.getProperty('confluenceUser')}:${project.getProperty('confluencePass')}".bytes.encodeBase64().toString() 
+    config.confluence.credentials = "${project.getProperty('confluenceUser')}:${project.getProperty('confluencePass')}".bytes.encodeBase64().toString()
 }
 if (project.hasProperty('username') && project.hasProperty('password')) {
     logger.info("Found passed common Jira & Confluence credentials")
-    config.jira.credentials = "${project.getProperty('username')}:${project.getProperty('password')}".bytes.encodeBase64().toString() 
-    config.confluence.credentials = "${project.getProperty('username')}:${project.getProperty('password')}".bytes.encodeBase64().toString() 
+    config.jira.credentials = "${project.getProperty('username')}:${project.getProperty('password')}".bytes.encodeBase64().toString()
+    config.confluence.credentials = "${project.getProperty('username')}:${project.getProperty('password')}".bytes.encodeBase64().toString()
 }
 
 inputPath = config.inputPath?config.inputPath:'.'
 referenceDocFile = config.referenceDocFile?config.referenceDocFile:''
 logger.info "docToolchain> referenceDocFile: ${referenceDocFile}"
 logger.info("\n==================\nParsed config file has ${config.size()} entries\n")
-config.each {key, value -> 
+config.each {key, value ->
     logger.info("Found config -> '${key}': '${value}'")
 }
 
@@ -143,6 +143,10 @@ task generateHTML (
             'html' in it.formats
         }.empty
     }
+
+    // specify output folder explicitly to avoid cleaning targetDir from other generated content
+    outputDir = file(targetDir + '/html5/')
+    separateOutputDirs(false)
 
     sources {
         sourceFiles.findAll {

--- a/scripts/exportContributors.gradle
+++ b/scripts/exportContributors.gradle
@@ -6,10 +6,6 @@ task exportContributors(
         description: 'exports all contributors for all asciidoc files',
         group: 'docToolchain'
 ) {
-    doFirst {
-        new File(targetDir, 'contributors').mkdirs()
-        new File(targetDir, 'fileattribs').mkdirs()
-    }
     doLast {
         def wordsPerMinute = 189
         def md5 = { input ->
@@ -38,22 +34,22 @@ task exportContributors(
                                 }
                 }.unique { a, b -> a[1] <=> b[1] }
                 def avatars = contributors.collect {
-                    "image:http://www.gravatar.com/avatar/" +
+                    "image:https://www.gravatar.com/avatar/" +
                             md5(it[1].toLowerCase().trim()) +
                             "?d=identicon" +
-                            "[32,32,role='gravatar',alt='${it[0]}',title='${it[0]}']"
+                            "[height=32,width=32,role='gravatar',alt='${it[0]}',title='${it[0]}']"
                 }.join(" ")
                 def tag = (file.canonicalPath - sourceFolder.canonicalPath).replace("\\", "/").replaceAll("^/", "")
-                def targetFile = new File(targetDir, "/contributors/" + tag)
-                new File(targetDir, ("/contributors/" + tag).split("/")[0..-2].join("/")).mkdirs()
+                File targetFile = new File(targetDir.toString(), "/contributors/" + tag)
+                safeMkDirs(targetFile)
                 targetFile.write(avatars + "\n")
 //              println("\u023F1 $readingtime minutes to read")
                 def minutesTerm = readingtime == 1 ? "minute" : "minutes"
                 targetFile.append("\n $readingtime $minutesTerm to read \n")
 
                 //file attributes
-                new File(targetDir, ("/fileattribs/" + tag).split("/")[0..-2].join("/")).mkdirs()
-                targetFile = new File(targetDir, "/fileattribs/" + tag)
+                targetFile = new File(targetDir.toString(), "/fileattribs/" + tag)
+                safeMkDirs(targetFile)
 
                 def attributes = ["git", "log", "-1", "--no-merges", "--pretty=format:%x7c%x20%at%n%x7c%x20%ad%x20%n%x7c%x20%aE%n%x7c%x20%an%x20%n%x7c%x20%s%x20%n", "--date=short", "HEAD", currentFileName].execute()
                 attributes.waitForOrKill(1000)
@@ -90,6 +86,15 @@ def getReadingtime(def filename, def wpm) {
                     .split("\\s+");
 
     return (words.length / wpm).toInteger() +1
+}
+
+void safeMkDirs(File file) {
+    File fileDir = file.getParentFile();
+    if (!fileDir.exists()) {
+        if (!fileDir.mkdirs()) {
+            throw new RuntimeException("unable to create folder '" + fileDir + '')
+        }
+    }
 }
 
 //end::exportContributors[]

